annotation/tests/test_jobs.py

```diff
@@ -497,3 +505,363 @@ def test_update_user_overall_load(tasks: Tuple[ManualAnnotationTask]):
+        )
+        mock_recalculate_file_pages.assert_called_with(
+            mock_session, expected_recalculate_calls
+        )
+        mock_update_user_overall_load.assert_has_calls(
+            expected_user_load_calls, any_order=True
+        )
+
+
+@pytest.mark.parametrize(
+    (
+        "new_validators",
+        "new_annotators",
+        "job_validators",
+        "job_annotators",
+        "expected_annotators",
+        "expected_validators",
+    ),
+    (
+        (
+            {UUID(int=1), UUID(int=2)},
+            {UUID(int=3)},
+            (User(user_id=UUID(int=1)), User(user_id=UUID(int=2))),
+            (User(user_id=UUID(int=3)), User(user_id=UUID(int=4))),
+            {UUID(int=4)},
+            set(),
+        ),
+        (
+            {UUID(int=3)},
+            {UUID(int=2), UUID(int=1)},
+            (User(user_id=UUID(int=4)), User(user_id=UUID(int=3))),
+            (User(user_id=UUID(int=2)), User(user_id=UUID(int=1))),
+            set(),
+            {UUID(int=4)},
+        ),
+    ),
+)
+def test_find_saved_users(
+    new_validators: Set[UUID],
+    new_annotators: Set[UUID],
+    job_validators: Tuple[User],
+    job_annotators: Tuple[User],
+    expected_annotators: Set[UUID],
+    expected_validators: Set[UUID],
+):
+    mock_session = MagicMock()
+
+    mock_job = MagicMock()
+    mock_job.validators = job_validators
+    mock_job.annotators = job_annotators
+    with patch(
+        "annotation.jobs.services.delete_tasks_for_removed_users",
+        return_value=[ManualAnnotationTask(user_id=UUID(int=4))],
+    ):
+        (
+            result_annotators,
+            result_validators,
+        ) = services.find_saved_users(
+            mock_session, mock_job, new_annotators, new_validators
+        )
+        assert result_validators == expected_validators
+        assert result_annotators == expected_annotators
+
+
+@pytest.mark.parametrize(
+    (
+        "patch_data",
+        "is_manual",
+        "expected_deleted",
+        "expected_annotators",
+        "expected_validators",
+    ),
+    (
+        (
+            {
+                "annotators": {UUID(int=5)},
+                "validators": {UUID(int=6)},
+                "owners": {UUID(int=7)},
+            },
+            True,
+            {UUID(int=1), UUID(int=2), UUID(int=3), UUID(int=4)},
+            set(),
+            set(),
+        ),
+        (
+            {
+                "annotators": {UUID(int=1)},
+                "validators": {UUID(int=2)},
+                "owners": {UUID(int=4)},
+            },
+            True,
+            {UUID(int=3)},
+            set(),
+            set(),
+        ),
+        (
+            {
+                "annotators": {UUID(int=1)},
+                "validators": {UUID(int=2)},
+                "owners": {UUID(int=4)},
+            },
+            False,
+            set(),
+            set(),
+            set(),
+        ),
+    ),
+)
```

Comment:
You're passing the same args in all 3 test cases. You could just add inside the function `expected_annotators = set()` and `expected_validators=set()` and you wouldn't need them in the parametrize